modules/nf-core/cutadapt/cutadapt.diff

```diff
@@ -0,0 +1,54 @@
+Changes in module 'nf-core/cutadapt'
+--- modules/nf-core/cutadapt/main.nf
++++ modules/nf-core/cutadapt/main.nf
+@@ -2,17 +2,17 @@
+     tag "$meta.id"
+     label 'process_medium'
+ 
+-    conda "bioconda::cutadapt=3.4"
++    conda 'bioconda::cutadapt=3.4'
+     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+         'https://depot.galaxyproject.org/singularity/cutadapt:3.4--py39h38f01e4_1' :
+         'quay.io/biocontainers/cutadapt:3.4--py39h38f01e4_1' }"
+ 
+     input:
+-    tuple val(meta), path(reads)
++    tuple val(meta), path(adapter_seq), path(reads)
+ 
+     output:
+-    tuple val(meta), path('*.trim.fastq.gz'), emit: reads
+-    tuple val(meta), path('*.log')          , emit: log
++    tuple val(meta), path('*.trim.fastq.gz'), optional: true, emit: reads
++    tuple val(meta), path('*.log')          , optional: true, emit: log
+     path "versions.yml"                     , emit: versions
+ 
+     when:
+@@ -21,12 +21,12 @@
+     script:
+     def args = task.ext.args ?: ''
+     def prefix = task.ext.prefix ?: "${meta.id}"
+-    def trimmed  = meta.single_end ? "-o ${prefix}.trim.fastq.gz" : "-o ${prefix}_1.trim.fastq.gz -p ${prefix}_2.trim.fastq.gz"
++    if (reads != [])
```

Comment:
Correct me if I understood wrongly, but why would you use cutadapt when no reads?

Reply:
We are running cutadapt if there are reads. 
In the other case, the variable reads will be empty, this input channel comes from a fastqc step which detects overrepresented sequences (here named adapter_seq), if adapters are not found the input channel contains only meta and reads. 
It's true that this is a bit confusing, maybe changing the order of the channel is the best.